libpromises/evalfunction.c

```diff
@@ -2066,6 +2067,157 @@ static FnCallResult FnCallClassify(EvalContext *ctx,
     return FnReturnContext(is_defined);
 }
 
+/*********************************************************************/
+
+static VersionComparison GenericVersionCheck(
+        const FnCall *fp,
+        const Rlist *args)
+{
+    assert(fp != NULL);
+    assert(fp->name != NULL);
+    if (args == NULL)
+    {
+        Log(LOG_LEVEL_ERR,
+            "Policy fuction %s requires version to compare against",
+            fp->name);
+        return VERSION_ERROR;
+    }
```

Comment:
Very nice, this helper function makes the code very easy to read and follow. 👏 👏 👏